mitiq/zne/inference.py

```diff
@@ -223,10 +223,15 @@ def __init__(self) -> None:
         self._options: Dict[str, Optional[float]] = {}
 
     def get_scale_factors(self) -> List[float]:
-        """Returns the scale factors at which the factory has computed
-        expectation values.
+        """Returns the scale factors that were either passed in to the facotry
+        or at which the factory has computed expectation values.
         """
-        return [params.get("scale_factor", 0.0) for params in self._instack]
+        ret_scale_factors = [
+            params.get("scale_factor", 0.0) for params in self._instack
+        ]
+        if not ret_scale_factors and hasattr(self, "_scale_factors"):
+            return [scale_factor for scale_factor in self._scale_factors]
+        return ret_scale_factors
```

Comment:
1. Can we just return `self._scale_factors` if it exists, rather than using the list comprehension? I have a feeling this was left over from when `float` was being called.
2. `ret_scale_factors` -> `scale_factors` as a variable name, I think is more clear.